extensions/sql-database-projects/src/models/project.ts

```diff
@@ -973,13 +985,48 @@ export class Project implements ISqlProject {
 		this.findOrCreateItemGroup(constants.Folder).appendChild(newFolderNode);
 	}
 
-	private removeFolderFromProjFile(path: string): void {
+	private async removeFolderFromProjFile(folderPath: string): Promise<void> {
 		const folderNodes = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Folder);
-		const deleted = this.removeNode(path, folderNodes);
+		let deleted = this.removeNode(folderPath, folderNodes);
 
+		// if it wasn't deleted, try deleting the folder path without trailing backslash
+		// since sometimes SSDT adds folders without a trailing \
 		if (!deleted) {
-			throw new Error(constants.unableToFindObject(path, constants.folderObject));
+			deleted = this.removeNode(utils.trimChars(folderPath, '\\'), folderNodes);
 		}
+
+		if (this.isSdkStyleProject) {
```

Comment:
Will it break if we don't perform this check? If user converts to SDK-style after this, the folder would be included again right? If adding the XML works for non-SDK-style projects, maybe we should just do it anyways so the user don't have to exclude it again later when they switch. Thoughts?

Reply:
And this applies to other checks for SDK-style projects too where we change the XML

Reply:
True that's a good point. Since this more of a migration scenario, I'll add a TODO comment to get back to this when we discuss the migration scenario in January.